source

```diff
@@ -99859,6 +99859,36 @@ dictionary <dfn>StorageEventInit</dfn> : <span>EventInit</span> {
      <th>Description
     <tbody>
     <tr>
+     <td><dfn data-x="parse-error-abandoned-head-element-child">abandoned-head-element-child</dfn>
+     <td><p>This error occurs if the parser encounters a <span data-x="syntax-start-tag">start
+     tag</span> for an element that is allowed by the parsing rules to be a <code>head</code>
+     element's child, but placed between the <code>head</code> element's
+     <span data-x="syntax-end-tag">end tag</span> and the <code>body</code> element's start tag or
+     any content that opens the <code>body</code> element if it's start tag is <span
+     data-x="syntax-tag-omission">omitted</span>. The parser treats such elements as child elements
+     of <code>head</code> element.</p>
+
+     <div class="example">
+       <p>For example, consider the following markup:</p>
+
+       <pre>&lt;!DOCTYPE html>
+&lt;html>
+ &lt;head>&lt;/head>
+ &lt;!-- This is a comment -->
+ &lt;title>The Answer&lt;/title>
+ &lt;script src="foo.js">&lt;/script>
+ &lt;body>&lt;/body>
+&lt;/html></pre>
+
+       <p>This will be parsed into:</p>
+
+       <ul class="domTree"><li class="t10">DOCTYPE: <code>html</code></li><li class="t1"><code>html</code><ul><li class="t1"><code>head</code><ul><li class="t1"><code>title</code><ul><li class="t3"><code>#text</code>: <span data-x="">The Answer</span></li></ul></li><li class="t1"><code>script</code> <span class="t2"><code class="attribute name" data-x="attr-id">src</code>="<code class="attribute value" data-x="">foo.js</code>"</span></li></ul><li class="t8"><code>#comment</code>: <span data-x=""> This is a comment </span></li><li class="t1"><code>body</code></li></ul></li></ul>
```

Comment:
Missing text nodes (with whitespace data) before and after the comment node, given the input markup.

Reply:
I guess I'll better remove comment node from example: it will add too much visual noise to the resulting tree without having any crucial explanatory role.